Remove unnecessary check in `TreeResolver::resolvePseudoElement()`

obyknovenius opened the pull request:
#### ad52c1163560b7c2756bdbbbfb4ae9318d420a90
<pre>
Remove unnecessary check in `TreeResolver::resolvePseudoElement()`
<a href="https://bugs.webkit.org/show_bug.cgi?id=257641">https://bugs.webkit.org/show_bug.cgi?id=257641</a>

Reviewed by Antti Koivisto.

There was a `FIXME` saying that this test shouldn&apos;t be needed.

* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/marker-computed-content-expected.txt:
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolvePseudoElement):

Canonical link: <a href="https://commits.webkit.org/265318@main">https://commits.webkit.org/265318@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4569a0d62eb5f990728a107a12added492b00264

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10545 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10769 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11046 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12191 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10117 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13138 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13052 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10706 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11636 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8868 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12592 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8930 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9524 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16786 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/8914 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10001 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9676 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12929 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10002 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10136 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8219 "Passed tests") | [❌ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/10667 "Failed to compile JSC") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9292 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/9418 "Passed tests") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/10667 "Failed to compile JSC") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2533 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13554 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/10964 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9994 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2690 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/rendering/updating/RenderTreeUpdaterGeneratedContent.cpp:
```diff
@@ -200,8 +200,8 @@ void RenderTreeUpdater::GeneratedContent::updateBackdropRenderer(RenderElement&
-    auto style = renderer.getCachedPseudoStyle(PseudoId::Backdrop, &renderer.view().style());
+    // ::backdrop does not inherit style
+    auto style = renderer.getCachedPseudoStyle(PseudoId::Backdrop, nullptr);
     if (!style || style->display() == DisplayType::None) {
```
These changes seem to be causing regressions according to EWS

obyknovenius replied:
I wasn't it but I reverted it back.

anttijk commented on Source/WebCore/style/StyleTreeResolver.cpp:
```diff
@@ -504,11 +495,13 @@ ResolutionContext TreeResolver::makeResolutionContext()
                 return firstLineStyle;
         }
+        if (pseudoId == PseudoId::Backdrop)
+            return &m_parentStack.first().style;
```
I don't understand this change. It is not mentioned in the commit message either. Why is it needed? Where is it tested?

obyknovenius replied:
I reworked this part and made a separate PR #14915. I' will update this PR when that one it will be landed.

carlosgcampos committed:
REGRESSION(264826@main): [GStreamer] Fill information on the gst GL context
https://bugs.webkit.org/show_bug.cgi?id=258199

Reviewed by Miguel Gomez.

In 264826@main I removed the activate and fill of the gst GL context,
confused by the API docs of gst_gl_context_fill_info() that says that
it's typically used with wrapped contexts to allow wrapped contexts to
be used as regular GstGLContexts. We don't want it to be used as a
regular context, but the regular context uses the filled information of
the sharing context to get the EGL config.

* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::unmakeContextCurrent):
(WebCore::GLContext::ScopedGLContextCurrent::ScopedGLContextCurrent):
(WebCore::GLContext::ScopedGLContextCurrent::~ScopedGLContextCurrent):
* Source/WebCore/platform/graphics/egl/GLContext.h:
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
(WebCore::PlatformDisplay::gstGLContext const):

Canonical link: https://commits.webkit.org/265302@main

jjgriego committed:
Fix Mac JSCOnly build
https://bugs.webkit.org/show_bug.cgi?id=257720

Reviewed by Elliott Williams.

The build currently fails because this header is included from wtf/Assertions.h
but is not copied as a public header in the cmake build; an easy fix, and I
think it should be appropriate (only changed for this port)

* Source/WTF/wtf/PlatformJSCOnly.cmake:

Canonical link: https://commits.webkit.org/265303@main

alanbaradlay committed:
[FFC] FlexLayout::handleCrossAxisAlignmentForFlexItems should follow spec language
https://bugs.webkit.org/show_bug.cgi?id=258246

Reviewed by Antti Koivisto.

This patch implements https://www.w3.org/TR/css-flexbox-1/#cross-alignment
(Resolve cross axis auto margins + align flex items driven by align-self and align-content)

* Source/WebCore/layout/formattingContexts/flex/FlexLayout.cpp:
(WebCore::Layout::FlexLayout::handleCrossAxisAlignmentForFlexItems const):

Canonical link: https://commits.webkit.org/265304@main

alanbaradlay committed:
[FFC] FlexLayout::handleCrossAxisAlignmentForFlexLines should follow spec language
https://bugs.webkit.org/show_bug.cgi?id=258247

Reviewed by Antti Koivisto.

This patch implements https://www.w3.org/TR/css-flexbox-1/#algo-line-align
(Align flex lines based on 'align-content')

* Source/WebCore/layout/formattingContexts/flex/FlexLayout.cpp:
(WebCore::Layout::FlexLayout::handleCrossAxisAlignmentForFlexLines const):

Canonical link: https://commits.webkit.org/265305@main

alanbaradlay committed:
[FFC] Add skeleton implementation for layout integration dependent functions (flexContainerMainSize/hypotheticalCrossSizeForFlexItems/collapseNonVisibleFlexItems)
https://bugs.webkit.org/show_bug.cgi?id=258248

Reviewed by Antti Koivisto.

These functions trigger layout on flex items while running flex layout and they need integration support (calling back in to render tree).
However currently FFC only works with "fixed" geometry meaning that we don't really need to run layout on flex items' subtrees.

* Source/WebCore/layout/formattingContexts/flex/FlexLayout.cpp:
(WebCore::Layout::FlexLayout::flexContainerMainSize const):
(WebCore::Layout::FlexLayout::hypotheticalCrossSizeForFlexItems const):
(WebCore::Layout::FlexLayout::collapseNonVisibleFlexItems):

Canonical link: https://commits.webkit.org/265306@main

TingPing committed:
[GLIB] Allow developer permissions in bwrap sandbox with env var
https://bugs.webkit.org/show_bug.cgi?id=257809

Reviewed by Michael Catanzaro.

This allows using tools like gdb, rr, or anything relying on ptrace
in the sandbox.

* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
(WebKit::setupSeccomp):
(WebKit::bubblewrapSpawn):

Canonical link: https://commits.webkit.org/265307@main

alanbaradlay committed:
[FFC] FlexLayout::computeAvailableMainAndCrossSpace should follow spec language
https://bugs.webkit.org/show_bug.cgi?id=258249

Reviewed by Antti Koivisto.

This patch implements https://www.w3.org/TR/css-flexbox-1/#algo-available
(Figure out how much available space there is for flex items in both directions)

* Source/WebCore/layout/formattingContexts/flex/FlexLayout.cpp:
(WebCore::Layout::FlexLayout::computeAvailableMainAndCrossSpace):

Canonical link: https://commits.webkit.org/265308@main

anttijk committed:
Limit maximum CSS rule nesting level
https://bugs.webkit.org/show_bug.cgi?id=258272
rdar://110312738

Reviewed by Alan Baradlay.

Deeply nested content (like accidentally unterminated @media rule lists) may end up blowing up the stack.

* LayoutTests/fast/media/mq-nesting-limit-expected.txt: Added.
* LayoutTests/fast/media/mq-nesting-limit.html: Added.
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeRegularRuleList):

Add a limit of 128 levels of nesting.

(WebCore::CSSParserImpl::consumeStyleRule):
* Source/WebCore/css/parser/CSSParserImpl.h:
(WebCore::CSSParserImpl::isNestedContext):

Canonical link: https://commits.webkit.org/265309@main

litherum committed:
[JPEGXL] Migrate JXLDecoderPtr to be a std::unique_ptr
https://bugs.webkit.org/show_bug.cgi?id=258283
rdar://111008382

Reviewed by Cameron McCormack.

There's no need to duplicate the functionality of std::unique_ptr.

* Source/WebCore/platform/image-decoders/jpegxl/JxlDecoderPtr.h:
(JXLDecoderDeleter::operator()):
(JxlDecoderMake):
(JxlDecoderPtr::JxlDecoderPtr): Deleted.
(JxlDecoderPtr::operator=): Deleted.
(JxlDecoderPtr::~JxlDecoderPtr): Deleted.
(JxlDecoderPtr::reset): Deleted.
(JxlDecoderPtr::operator bool const): Deleted.
(JxlDecoderPtr::get const): Deleted.
(): Deleted.

Canonical link: https://commits.webkit.org/265310@main

jyavenard committed:
CrashTracer: com.apple.WebKit.WebContent at WebCore: WTF::Detail::CallableWrapper<WebCore::FullscreenManager::cancelFullscreen()::$_5, void>::call
https://bugs.webkit.org/show_bug.cgi?id=258268
rdar://107182009

Reviewed by Youenn Fablet and Tim Nguyen.

Exit early if either FullscreenManager has been deleted (not possible, but may as well check).
And use the topDocument's page to trigger exitFullscreen.

* LayoutTests/fullscreen/fullscreen-iframe-cancel-expected.txt: Added.
* LayoutTests/fullscreen/fullscreen-iframe-cancel.html: Added.
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::cancelFullscreen): Use topDocument's page instead of page.

Canonical link: https://commits.webkit.org/265311@main

tadeuzagallo committed:
[WGSL] Type checker crashes when calling a non-existing function
https://bugs.webkit.org/show_bug.cgi?id=258280
rdar://111003956

Reviewed by Dan Glastonbury.

When type checking a call we try to identify what is the type bound to the target,
e.g. for `f()`, first we look up the typing context to see if there's a type or
value bound to `f`. If there is none, we try to look up overloaded declarations
for `f`. If `f` isn't overloaded either, we have to emit an error, but at this
point we were unconditionally using the binding to construct the error message
(the first lookup just mentioned), which will be null if `f` doesn't exist. To
fix that we check if the binding exists to decide which error message we should
emit: either that the target doesn't exist or that it's an invalid target for a
call (i.e. it exists, but isn't a function).

* Source/WebGPU/WGSL/TypeCheck.cpp:
(WGSL::TypeChecker::visit):
* Source/WebGPU/WGSL/tests/invalid/function-call.wgsl:

Canonical link: https://commits.webkit.org/265312@main

lukewarlow committed:
Implement scrollbar-color CSS parsing
https://bugs.webkit.org/show_bug.cgi?id=257571

Reviewed by Darin Adler and Tim Nguyen.

Spec: https://drafts.csswg.org/css-scrollbars/#scrollbar-color

Grammar for property: auto | <color>{2}

This adds the property to the CSS parser behind a flag.
This also adds animation support for this property.
Test expectations have been updated, and new tests have been added.

* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-color-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-types.js:
(const.colorPairType.testInterpolation):
(const.colorPairType.testAdditionOrAccumulation):
(const.colorPairType.testAddition):
(const.colorPairType.testAccumulation):
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeScrollSnapType):
(WebCore::CSSPropertyParserHelpers::consumeScrollbarColor):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::initialScrollbarColor):
(WebCore::RenderStyle::scrollbarColor const):
(WebCore::RenderStyle::scrollbarThumbColor const):
(WebCore::RenderStyle::scrollbarTrackColor const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setScrollbarColor):
(WebCore::RenderStyle::setScrollbarThumbColor):
(WebCore::RenderStyle::setScrollbarTrackColor):
* Source/WebCore/rendering/style/ScrollbarColor.cpp: Added.
(WebCore::operator<<):
* Source/WebCore/rendering/style/ScrollbarColor.h: Added.
(WebCore::ScrollbarColor::MarkableTraits::isEmptyValue):
(WebCore::ScrollbarColor::MarkableTraits::emptyValue):
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
(WebCore::StyleRareInheritedData::StyleRareInheritedData):
(WebCore::StyleRareInheritedData::operator== const):
* Source/WebCore/rendering/style/StyleRareInheritedData.h:
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertScrollbarColor):

Canonical link: https://commits.webkit.org/265313@main

alanbaradlay committed:
[FFC] Populate LogicalFlexItem::Main and CrossAxisGeometry for flex layout
https://bugs.webkit.org/show_bug.cgi?id=258252

Reviewed by Antti Koivisto.

* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.cpp:
(WebCore::Layout::FlexFormattingContext::convertFlexItemsToLogicalSpace):

Canonical link: https://commits.webkit.org/265314@main

twilco committed:
AX: AXTextMarker::boundaryPoint() can nullptr crash after WeakPtr<Node> is destroyed
https://bugs.webkit.org/show_bug.cgi?id=258291
rdar://110921085

Reviewed by Chris Fleizach.

After creating a `WeakPtr` to a `Node`, `AXTextMarker::boundaryPoint()` calls `AccessibilityObject::replacedNodeNeedsCharacter`
with that node, which in turn calls `accessibilityIsIgnored`, which can cause side effects that destroy the original `WeakPtr` `Node`.
We need to re-check that the `Node` didn't become nullptr (get destroyed) after calling `AccessibilityObject::replacedNodeNeedsCharacter`.

* Source/WebCore/accessibility/AXTextMarker.cpp:
(WebCore::AXTextMarker::boundaryPoint const):

Canonical link: https://commits.webkit.org/265315@main

philn committed:
[GStreamer][MediaStream] Move mock display source to its own cpp file
https://bugs.webkit.org/show_bug.cgi?id=258072

Reviewed by Xabier Rodriguez-Calvar.

Moving code around for consistency purposes.

* Source/WebCore/platform/SourcesGStreamer.txt:
* Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.cpp: Copied from Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp.
(WebCore::MockDisplayCaptureSourceGStreamer::create):
(WebCore::MockDisplayCaptureSourceGStreamer::MockDisplayCaptureSourceGStreamer):
(WebCore::MockDisplayCaptureSourceGStreamer::~MockDisplayCaptureSourceGStreamer):
(WebCore::MockDisplayCaptureSourceGStreamer::stopProducingData):
(WebCore::MockDisplayCaptureSourceGStreamer::requestToEnd):
(WebCore::MockDisplayCaptureSourceGStreamer::setMuted):
(WebCore::MockDisplayCaptureSourceGStreamer::videoFrameAvailable):
(WebCore::MockDisplayCaptureSourceGStreamer::capabilities):
(WebCore::MockDisplayCaptureSourceGStreamer::settings):
* Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.h: Copied from Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.h.
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp:
(WebCore::MockDisplayCaptureSourceGStreamer::create): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::MockDisplayCaptureSourceGStreamer): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::~MockDisplayCaptureSourceGStreamer): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::stopProducingData): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::requestToEnd): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::setMuted): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::videoFrameAvailable): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::capabilities): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::settings): Deleted.
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.h:
* Source/WebCore/platform/mock/MockRealtimeMediaSourceCenter.cpp:

Canonical link: https://commits.webkit.org/265316@main

pgorszkowski-igalia committed:
[GLIB] Add API to control originStorageRatio and totalStorageRatio
https://bugs.webkit.org/show_bug.cgi?id=254905

Reviewed by Carlos Garcia Campos.

Add API to control originStorageRatio and totalStorageRatio.

* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
(webkitWebsiteDataManagerSetProperty):
(webkit_website_data_manager_class_init):
(webkitWebsiteDataManagerGetDataStore):

Canonical link: https://commits.webkit.org/265317@main

anttijk commented on Source/WebCore/style/StyleTreeResolver.cpp:
```diff
@@ -339,15 +339,6 @@ std::optional<ElementUpdate> TreeResolver::resolvePseudoElement(Element& element
     if (!resolvedStyle)
         return { };
 
-    // FIXME: This test shouldn't be needed.
-    bool alwaysNeedsPseudoElement = resolvedStyle->style->hasAnimationsOrTransitions()
-        || element.hasKeyframeEffects(pseudoId)
-        || pseudoId == PseudoId::FirstLine
-        || pseudoId == PseudoId::FirstLetter
-        || pseudoId == PseudoId::Scrollbar;
-    if (!alwaysNeedsPseudoElement && !pseudoElementRendererIsNeeded(resolvedStyle->style.get()))
-        return { };
-
```
nice

obyknovenius committed:
Remove unnecessary check in `TreeResolver::resolvePseudoElement()`
https://bugs.webkit.org/show_bug.cgi?id=257641

Reviewed by Antti Koivisto.

There was a `FIXME` saying that this test shouldn't be needed.

* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/marker-computed-content-expected.txt:
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolvePseudoElement):

Canonical link: https://commits.webkit.org/265318@main
